tests/test_esmda.py

```diff
@@ -25,9 +25,55 @@
 
 from iterative_ensemble_smoother.esmda import ESMDA
 from iterative_ensemble_smoother.esmda_inversion import empirical_cross_covariance
+from iterative_ensemble_smoother.sies import SIES
 
 
 class TestESMDA:
+    @pytest.mark.parametrize("num_inputs", [10, 25, 50])
+    @pytest.mark.parametrize("num_outputs", [5, 25, 50])
+    @pytest.mark.parametrize("sies_inversion", ["direct", "subspace_exact"])
+    @pytest.mark.parametrize("seed", list(range(10)))
+    def test_that_ESMDA_and_SIES_produce_same_result_with_one_step(
+        self, seed, sies_inversion, num_outputs, num_inputs
+    ):
+        """With a single step (alpha=1), ESMDA = SIES = ES.
+
+        When num_inputs < num_ensemble - 1, then Section (2.4.3) in the SIES
+        paper triggers and the result is not identical.
+        """
+        rng = np.random.default_rng(seed)
+
+        num_ensemble = 10
+        alpha = 1
+
+        # Create problem instance
+        X = rng.normal(size=(num_inputs, num_ensemble))
+        Y = rng.normal(size=(num_outputs, num_ensemble))
+        covariance = np.exp(rng.normal(size=num_outputs))
+        observations = rng.normal(size=num_outputs, loc=1)
+
+        # Create ESMDA instance and perform one iteration
+        esmda = ESMDA(
+            covariance, observations, alpha=alpha, seed=seed + 99, inversion="exact"
+        )
+        X_ESMDA = np.copy(X)
+        for _ in range(esmda.num_assimilations()):
```

Comment:
Do we need this for-loop or is it enough to just call `assimilate` once?

Reply:
in this particular test it's not needed. I can remove it :)